python/tchannel/tornado/connection.py

```diff
@@ -593,19 +605,39 @@ def send_request(self, request):
         # TODO: fire before_receive_response
 
         def adapt_tracing(f):
```

Comment:
This method is doing way more than tracing, and it's complicated enough that it shouldn't be defined inline.